dislib/data/base.py

```diff
@@ -1,238 +1,149 @@
-import os
+from math import ceil
 
 import numpy as np
-from pycompss.api.parameter import FILE_IN
+from pycompss.api.parameter import Type, COLLECTION_INOUT, Depth
 from pycompss.api.task import task
-from scipy.sparse import issparse
 
-from dislib.data import Subset, Dataset
+from dislib.data.array import Array
 
 
-def load_data(x, subset_size, y=None):
-    """
-    Loads data into a Dataset.
-
-    Parameters
-    ----------
-    x : ndarray, shape=[n_samples, n_features]
-        Array of samples.
-    y : ndarray, optional, shape=[n_features,]
-        Array of labels.
-    subset_size : int
-        Subset size in number of samples.
-
-    Returns
-    -------
-    dataset : Dataset
-        A distributed representation of the data divided in Subsets of
-        subset_size.
-    """
-    dataset = Dataset(n_features=x.shape[1], sparse=issparse(x))
-
-    for i in range(0, x.shape[0], subset_size):
-        if y is not None:
-            subset = Subset(x[i: i + subset_size], y[i: i + subset_size])
-        else:
-            subset = Subset(x[i: i + subset_size])
-        dataset.append(subset)
-
-    return dataset
-
-
-def load_libsvm_file(path, subset_size, n_features, store_sparse=True):
-    """ Loads a LibSVM file into a Dataset.
+def load_svmlight_file(path, block_size, n_features, store_sparse):
```

Comment:
I thought this file would be deleted once the array was finished. The methods defined here are used to construct arrays. However, the array.py also has methods for constructing array (array(), random_array()). I think all methods should be either inside arra.py or in this file but not divided.